block/manager.go

```diff
@@ -150,18 +174,28 @@ func NewManager(
 		retriever:   dalc.(da.BlockRetriever), // TODO(tzdybal): do it in more gentle way (after MVP)
 		daHeight:    s.DAHeight,
 		// channels are buffered to avoid blocking on input/output operations, buffer sizes are arbitrary
-		HeaderCh:          make(chan *types.SignedHeader, 100),
-		BlockCh:           make(chan *types.Block, 100),
-		blockInCh:         make(chan newBlockEvent, 100),
-		retrieveMtx:       new(sync.Mutex),
-		lastStateMtx:      new(sync.RWMutex),
-		syncCache:         make(map[uint64]*types.Block),
+		HeaderCh:      make(chan *types.SignedHeader, channelLength),
+		BlockCh:       make(chan *types.Block, channelLength),
+		blockInCh:     make(chan newBlockEvent, channelLength),
+		blockStoreMtx: new(sync.Mutex),
+		blockStore:    blockStore,
+		retrieveMtx:   new(sync.Mutex),
+		lastStateMtx:  new(sync.RWMutex),
+		blockCache: &BlockCache{
```

Comment:
[non-blocking] might as well and a `newBlockCache` method to refactor this to keep everything contained.